flexget/utils/tools.py

```diff
@@ -14,58 +14,171 @@
 import os
 import re
 import sys
+import functools
+import queue
+import requests
 from collections import MutableMapping
 from datetime import timedelta, datetime
 from pprint import pformat
+from dateutil.parser import parse as dateutil_parse
+from dateutil.tz import tz
+from time import mktime
+from html.entities import name2codepoint
 
 import flexget
-import queue
-import requests
 
-from html.entities import name2codepoint
 
 log = logging.getLogger('utils')
 
 
-def str_to_boolean(string):
-    return string.lower() in ['true', '1', 't', 'y', 'yes']
+def str_to_boolean(value):
+    return value.lower() in ['true', '1', 't', 'y', 'yes']
 
 
-def str_to_int(string):
+def str_to_int(value):
     try:
-        return int(string.replace(',', ''))
+        return int(re.sub(r'[,.]', '', value))
     except ValueError:
         return None
 
 
+def str_to_number(value):
+    try:
+        return float(value)
+    except ValueError:
+        return str_to_int(value)
+
+
+def value_to_number(value):
+    if isinstance(value, str):
+        return str_to_number(value)
+    elif isinstance(value, (int, float)):
+        return value
+    elif isinstance(value, bool):
+        return int(value)
+    else:
+        return None
+
+
+def str_to_naive_utc(value):
+    parsed_date = None
+    try:
+        parsed_date = dateutil_parse(value, fuzzy=True)
+        try:
+            parsed_date = parsed_date.astimezone(tz.tzutc()).replace(tzinfo=None)
+        except ValueError:
+            parsed_date = parsed_date.replace(tzinfo=None)
+    except ValueError as ex:
+        log.warning('Invalid datetime field format: %s error: %s', value, ex)
+    return parsed_date
+
+
+def value_to_naive_utc(value):
```

Comment:
The `str_to` versions can only deal with string inputs, while the `value_to` versions can deal with different input types. I needed this since sometimes xml fields are already in some specific date/number format, instead of a string.